From 2990e46a90ace0074c4f6308814251132f947cae Mon Sep 17 00:00:00 2001
From: Jonas Gorski <jonas.gorski@bisdn.de>
Date: Mon, 7 Feb 2022 12:11:54 +0100
Subject: [PATCH 06/14] bcm-knet: report link state

We set netif carrier, so we can just use ethtool_op_get_link.

Signed-off-by: Jonas Gorski <jonas.gorski@bisdn.de>
---
 sdk-6.5.24/systems/linux/kernel/modules/bcm-knet/bcm-knet.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/sdk-6.5.24/systems/linux/kernel/modules/bcm-knet/bcm-knet.c b/sdk-6.5.24/systems/linux/kernel/modules/bcm-knet/bcm-knet.c
index 10d08f798ab0..e03de2b76fa3 100755
--- a/sdk-6.5.24/systems/linux/kernel/modules/bcm-knet/bcm-knet.c
+++ b/sdk-6.5.24/systems/linux/kernel/modules/bcm-knet/bcm-knet.c
@@ -6737,6 +6737,7 @@ bkn_get_ts_info(struct net_device *dev, struct ethtool_ts_info *info)
 
 static const struct ethtool_ops bkn_ethtool_ops = {
     .get_drvinfo        = bkn_get_drvinfo,
+    .get_link           = ethtool_op_get_link,
 #if (LINUX_VERSION_CODE >= KERNEL_VERSION(3,5,0))
     .get_ts_info        = bkn_get_ts_info,
 #endif
-- 
2.38.1

